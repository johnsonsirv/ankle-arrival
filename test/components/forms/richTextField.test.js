/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import RichTextField from '../../../src/components/forms/inputs/richTextField';

function setup() {
  const props = {
    name: 'description',
    onChange: jest.fn(),
  };
  const enzymeWrapper = mount(<RichTextField {...props} />);

  return { props, enzymeWrapper };
}

describe('<RichTextField /> rendering', () => {
  const { enzymeWrapper } = setup();
  it('renders correctly', () => {
    expect(enzymeWrapper).toMatchSnapshot();
  });
  it('should render a <textarea>', () => {
    expect(enzymeWrapper.find('textarea')).toHaveLength(1);
  });
});

describe('<RichTextField /> interactions', () => {
  const { props, enzymeWrapper } = setup();
  it('should handle onChange', () => {
    const rchText = enzymeWrapper.find('textarea');
    rchText.simulate('change', { target: { value: 'Sample Text' } });
    expect(props.onChange).toHaveBeenCalled();
  });
});
