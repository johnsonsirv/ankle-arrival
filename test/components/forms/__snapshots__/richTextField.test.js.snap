// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RichTextField /> rendering renders correctly 1`] = `ReactWrapper {}`;
